Pin openssl to 1.x for now and bump min macOS to 10.12

milljm opened the pull request:
We have an issue with secure renegotiation with our CI server. Pinning openssl will allow CI to continue.

Bumping the minimum macOS to Sierra (10.12) in order to fixup threading issues on Mac in framework and BISON

Closes #20668
Closes #20680

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999169/) on c0171fa wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999202/) on 0011ba3 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999202/) on 0011ba3 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented on conda/mpich/meta.yaml:
```diff
@@ -37,6 +37,7 @@ requirements:
     - automake                  # [osx]
     - libtool                   # [osx]
     - make                      # [osx]
+    - {{ pin_compatible('openssl', lower_bound='1', upper_bound='3') }}
```
OK, just making sure this doesn't pull version 3 - haven't looked up the logic here myself.

milljm replied:
I miss understood your comment... I thought you ment "you don't need the lower bounds"... 
But yes, upper_bounds is treated as less_than. Not less than or equal to.

milljm replied:
```yaml
requirements:
    build:
        - numpy {{ numpy }}
    run:
        - {{ pin_compatible('numpy', lower_bound='1.10', upper_bound='3.0') }}
```
This would yield a pinning of >=1.10,<3.0.


ref: https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999233/) on 8ebb3a7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented:
Rebase on next; we have another update sitting in there

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999289/) on cdc3eb0 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999289/) on cdc3eb0 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999289/) on cdc3eb0 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Pin openssl to 1.x for now

We have an issue with secure renegotiation with our CI server. This
change will allow CI to continue.

Closes #20668

Commit:
Without using jinja, limit openssl to less than v3

cticenhour commented on conda/mpich/meta.yaml:
```diff
@@ -37,7 +37,7 @@ requirements:
     - libtool                   # [osx]
     - make                      # [osx]
-    - {{ pin_compatible('setuptools', lower_bound='58', upper_bound='60') }}
+    - openssl <3
```
❤️

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999320/) on 93166e7 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999320/) on 93166e7 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999320/) on 93166e7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented:
`tapi` problem
```
Executing: mpicc  -o /tmp/petsc-_y_6kli7/config.setCompilers/conftest -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib  -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem $PREFIX/include   -march=core2 -mtune=haswell /tmp/petsc-_y_6kli7/config.setCompilers/conftest.o 
Possible ERROR while running linker: exit code 254
stderr:
dyld[63052]: Library not loaded: @rpath/libtapi.dylib
  Referenced from: /opt/civet0/build_0/conda_builds/conda_envs/pr-20669/.pkgs/ld64_osx-64-609-hc6b1d1e_10/bin/x86_64-apple-darwin13.4.0-ld
  Reason: tried: '/opt/civet0/build_0/conda_builds/conda_envs/pr-20669/.pkgs/ld64_osx-64-609-hc6b1d1e_10/bin/../lib/libtapi.dylib' (no such file), '/opt/civet0/build_0/conda_builds/conda_envs/pr-20669/.pkgs/ld64_osx-64-609-hc6b1d1e_10/bin/../lib/libtapi.dylib' (no such file), '/usr/local/lib/libtapi.dylib' (no such file), '/usr/lib/libtapi.dylib' (no such file)
clang-12: error: unable to execute command: Abort trap: 6
clang-12: error: linker command failed due to signal (use -v to see invocation)
```

Commit:
Forgot to bump libmesh-vtk in conda_build_config

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999378/) on 4664153 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999378/) on 4664153 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/999394/) on 4664153 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20669/site)

This comment will be updated on new commits.

cticenhour committed:
Bump minimum macOS SDK to 10.12 and increment libmesh build (#20680)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999757/) on 5d12084 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999757/) on 5d12084 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/999757/) on 5d12084 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/999781/) on 5d12084 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>824960</th>
      <th colspan=4>#20669 5d1208</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.58%</td>
      <td>82.58%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73938</td>
      <td>73935</td>
      <td>-3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15592</td>
      <td>15595</td>
      <td>+3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20669/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20669/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20669/coverage/xfem)

</details>



This comment will be updated on new commits.

cticenhour committed:
Pin ld64 in MPICH to re-gain control over tapi versioning (#20668)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000086/) on 9b2f396 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000086/) on 9b2f396 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000086/) on 9b2f396 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

tophmatthews commented:
Ay chihuahua, you guys have to do so much work behind the scenes to get these packages right. Thanks for kicking ass!

Commit:
More pinning, PETSc in own directory

Pin setuptools (again)
Install PETSc to specific directory

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000117/) on 2b2feca wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000117/) on 2b2feca wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000148/) on 5f02410 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000148/) on 5f02410 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Allow PETSC_DIR to remain CONDA_PREFIX

There are issues we need to work out with contribs, in order to make
this work. Revert back to allowing PETSc to install directly into
CONDA_PREFIX.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000330/) on 61791ff wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000330/) on 61791ff wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000330/) on 61791ff wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000624/) on 39b8a0e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000624/) on 39b8a0e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000624/) on 39b8a0e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000720/) on b26e593 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000720/) on b26e593 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000720/) on b26e593 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000780/) on ef43b44 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1000780/) on ef43b44 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour reviewed (approved):
Will merge despite end user mac silicon recipe; said recipe is a WIP and is not run in production. Just added for checking its WIP status.
